l10n_eu_product_adr/scripts/import_adr_multilang_xlsx.py

```diff
@@ -0,0 +1,485 @@
+        if un_number in by_qty:
+            parts = [un_number, qty]
+        else:
+            parts = [
+                un_number,
+                parse_classification_code(row),
+                "/".join(parse_labels(row)),
+                "/".join(parse_provisions_3_3(row)),
+                qty,
+                "/".join(parse_packing_instructions(row)),
+                parse_transport_category(row)[0],
+                apply_description_quirk(row),
+            ]
+    res = "_".join(part for part in parts if part).replace(".", "dot")
+    return "adr_goods_%s" % res
+
+
+def parse_un_number(row):
+    return str(row[index["un_number"]]).strip().zfill(4)
+
+
+@transformer
+def un_number(record, value, row):
+    xml_id = get_xml_id(row)
+    record.attrib.update(
+        {
+            "id": xml_id,
+            "model": "adr.goods",
+        }
+    )
+    etree.SubElement(
+        record, "field", attrib={"name": "un_number"}
+    ).text = parse_un_number(row)
+
+
+@transformer
+def packing_instruction_ids(record, value, row):
+    refs = []
+    for instruction in parse_packing_instructions(row):
+        refs.append("ref('adr_packing_instruction_%s')" % instruction)
+    expression = "[(6, 0, [%s])]" % ", ".join(refs)
+    etree.SubElement(
+        record,
+        "field",
+        attrib={
+            "name": "packing_instruction_ids",
+            "eval": expression,
+        },
+    )
+
+
+@transformer
+def limited_quantity(record, value, row):
+    quantity, uom = parse_limited_quantity(row)
+    if quantity:
+        etree.SubElement(
+            record, "field", attrib={"name": "limited_quantity"}
+        ).text = quantity
+        etree.SubElement(
+            record,
+            "field",
+            attrib={
+                "name": "limited_quantity_uom_id",
+                "ref": uom,
+            },
+        )
+
+
+@transformer
+def name(record, value, row):
+    value = value.strip().replace("\n", "")
+    etree.SubElement(record, "field", attrib={"name": "name"}).text = value
+
+
+@transformer
+def class_id(record, value, row):
+    etree.SubElement(
+        record,
+        "field",
+        attrib={
+            "name": "class_id",
+            "ref": "adr_class_%s" % value.replace(".", "_"),
+        },
+    )
+
+
+def parse_classification_code(row):
+    value = row[index["classification_code"]] or ""
+    # In case multiple codes are possible, take the first (rare)
+    code = value.split(" of ")[0]
+    return code.replace(",", ".").strip()  # Typo in the sheet
+
+
+@transformer
+def classification_code(record, value, row):
+    if value is None:  # Not defined for code 0190, for example
+        return
+    code = parse_classification_code(row)
+    etree.SubElement(
+        record, "field", attrib={"name": "classification_code"}
+    ).text = code
+
+
+def parse_transport_category(row):
+    value = row[index["transport_category"]]
+    if value is None:
+        if "VERVOER VERBODEN" in str(row):
```

Comment:
this is very customer specific

Reply:
This is not customer specific, it is specific to the sheet that contains the ADR data. It is made available by a Belgian organization, and as a consequence it contains a limited amount of Dutch terminology.